server/schema/company/company.datasources.js

```diff
@@ -17,12 +17,11 @@ const create = async (name, location, logo, session, authToken, mid) => {
   }
 };
 
-const find = async (query, limit, offset) =>
-  await CompanyModel.find(query).sort({ name: 'asc' }).skip(offset).limit(limit);
+const find = (query, limit, offset) => CompanyModel.find(query).sort({ name: 'asc' }).skip(offset).limit(limit);
 
 const findByID = () =>
   new DataLoader(
-    async (ids) => {
+    (ids) => {
       try {
         const _company = ids.map((id) => CompanyModel.findById(id));
```

Comment:
You have basically completely made it pointless to use dataloader with how your code runs.

Reply:
Data Loader is supposed to run a single query with all the ids, no run the query within a map this doing a lot of them individually.